test/extensions/filters/http/grpc_web/grpc_web_filter_test.cc

```diff
@@ -195,6 +195,25 @@ TEST_F(GrpcWebFilterTest, Base64NoPadding) {
   EXPECT_EQ(decoder_callbacks_.details(), "grpc_base_64_decode_failed_bad_size");
 }
 
+TEST_F(GrpcWebFilterTest, InvalidUpstreamResponseForText) {
+  Http::TestRequestHeaderMapImpl request_headers{
+      {"content-type", Http::Headers::get().ContentTypeValues.GrpcWebText}, {":path", "/"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "400"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_.encodeHeaders(response_headers, false));
+  Buffer::OwnedImpl data1("start");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer, filter_.encodeData(data1, false));
+  Buffer::OwnedImpl data2("hello");
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer, filter_.encodeData(data2, false));
+  Buffer::OwnedImpl data3("end");
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.encodeData(data3, true));
+
```

Comment:
can we verify response_headers has "grpc-message" with "start hello end"?

Reply:
My proposal will be: expect the last frame here, i.e. "end", since I think this is a failure anyway, so do not want to spend time buffering the content.

Reply:
SG